clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreIngestionTask.java

```diff
@@ -3034,6 +3041,7 @@ private int internalProcessConsumerRecord(
            * the last message of the sequence, which is not a chunk but rather the manifest.
            */
           validateMessage(
+              PartitionTracker.TopicType.VERSION_TOPIC,
               this.kafkaDataIntegrityValidator,
               consumerRecord,
```

Comment:
So if we want to truly separate RT tracking from VT tracking ,the `consumerRecord` needs to be from VT.
There was an issue we discussed in the past is that for hybrid store, the `consumerRecord` in leader replica is actually the one from RT topic instead of Version Topic, so I think we need to make some logic change in `leader replica` when producing to the local drainer queue in `LeaderProducerCallback`.

Reply:
Thanks a lot for your review! @gaojieliu . You are right that `consumerRecord` validated in here can come from both RT and VT topics, so it is inaccurate to put all of them into the VT DIV.

My thought is that this is the `drainer DIV` and its result is inaccurate because of the DCR and ignored messages. Based on the high level design of the snapshot approach, we need to later transfer the full DIV responsibility to the `consumer DIV` which resides in the consumer thread and then gradually retire the use of `drainer DIV`. However, because this the 1st PR for global RT series, DIV heartbeat is yet implemented, so I would like to keep `drainer DIV` the way as is today (a mixed DIV state). When future work is completed,  `drainer DIV` logic can be simply bypassed, and any work that we do today to fix the `drainer DIV` could be thrown away. This is the main reason I didn't change  `drainer DIV` logic. Do you think it makes sense?

Reply:
With today's logic, the VT DIV gap issue before reporting completed can still happen.
Let us leave a `TODO` here in case we forget it in the future.